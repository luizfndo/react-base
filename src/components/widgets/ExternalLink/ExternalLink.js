import React from 'react';
import PropTypes from 'prop-types';


function ExternalLink({
  to,
  alt,
  children,
  target,
  onClick,
}) {
  return (
    <a
      href={to}
      alt={alt}
      target={target}
      rel="noopener noreferrer"
      onClick={onClick}
    >
      {children}
    </a>
  );
}

ExternalLink.propTypes = {
  to: PropTypes.string.isRequired,
  alt: PropTypes.string,
  children: PropTypes.node.isRequired,
  target: PropTypes.string,
  onClick: PropTypes.func,
};

ExternalLink.defaultProps = {
  alt: '',
  target: '_blank',
  onClick: () => {},
};

export default ExternalLink;
