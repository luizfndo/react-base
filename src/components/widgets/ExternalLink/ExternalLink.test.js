import React from 'react';
import { create } from 'react-test-renderer';
import ExternalLink from './ExternalLink';


describe('ExternalLink component', () => {
  test('It matches the snapshot.', () => {
    const component = create(
      <ExternalLink to="/">My external link</ExternalLink>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
