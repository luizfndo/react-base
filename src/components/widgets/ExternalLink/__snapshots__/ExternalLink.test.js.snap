// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalLink component It matches the snapshot. 1`] = `
<a
  alt=""
  href="/"
  onClick={[Function]}
  rel="noopener noreferrer"
  target="_blank"
>
  My external link
</a>
`;
